app/models/product_reading.rb

```diff
@@ -90,6 +90,19 @@ class ProductReading < Ekylibre::Record::Base
     end
   end
 
+  after_save do
+    last_reading = product.readings.where(indicator_name: indicator_name).last
+    if last_reading.read_at <= read_at
+      product.send("#{indicator_name}=", value)
+      product.update_column(:reading_cache, product.reading_cache)
```

Comment:
This way to do can generate errors on concurrent calls

Reply:
Have you an idea of how we could avoid these errors? Should I add a semaphore call around it to ensure it won't get weird with concurrency ?